okhttp/src/main/kotlin/okhttp3/internal/connection/RealConnectionPool.kt

```diff
@@ -286,7 +303,82 @@ class RealConnectionPool(
     return references.size
   }
 
+  fun setPolicy(
+    address: Address,
+    policy: ConnectionPool.AddressPolicy,
+  ) {
+    // TODO: it's quite awkward that we have a new policy object that doesn't contain the max
+    require(policy.minimumConcurrentCalls <= maxIdleConnections) {
+      "minimumConcurrentCalls must be <= maxIdleConnections"
+    }
```

Comment:
The comment on `maxIdleConnections` says:

> The maximum number of idle connections for each address.

Which sounds like something that should also be a part of our new AddressPolicy.

Except, as far as I can tell from looking at the code, that's **not** what `maxIdleConnections` does. Instead, it seems to be a cap on the total number of connections in the pool, regardless of address.

So, how do we reconcile this?

Reply:
oldestConnectionsEvictedIfIdleLimitExceeded test confirms it's across addresses.

Reply:
I've updated the comment, and also updated `cleanup` to consider connections "in use" if they are needed to satisfy a policy